### ISSUE TITLE
Is possible to remove easyadmin-bundle top version limit?

### ISSUE THREAD

**Issue by rodrigobb** [alterphp/EasyAdminExtensionBundle#161]
EasyAdminBundle has several issues that has been fixed in v2.2 and v2.3 versions but I cannot update because EasyAdminExtensionBundle has a limit in `composer.json`: version of EasyAdminBundle <2.2.0.

I've made a fork of this bundle, removed the top version limit and made several quick tests and everything seems to work right. But I don't know how to run the complete test suite (maybe it's documented but I didn't find it) so I can't be sure.

Note: Do not update to EasyAdminBundle v2.3.1 if you are using CKEditor because it has an issue with it. v2.3.0 works well.

**Comment by andrea-daru** [comment]
> EasyAdminBundle has several issues that has been fixed in v2.2 and v2.3 versions but I cannot update because EasyAdminExtensionBundle has a limit in `composer.json`: version of EasyAdminBundle <2.0.0.
> 
> I've made a fork of this bundle, removed the top version limit and made several quick tests and everything seems to work right. But I don't know how to run the complete test suite (maybe it's documented but I didn't find it) so I can't be sure.
> 
> Note: Do not update to EasyAdminBundle v2.3.1 if you are using CKEditor because it has an issue with it. v2.3.0 works well.

Is it possible?

**Comment by alterphp** [comment]
Hello,

The reason of the top-limit are explained in #147.

List filters appeared in EasyAdmin with 2.2.0. They are not compatible with extension list filters. Embedded lists rely on extension list filters, so should remove embedded lists and remove top-limit. If you do so, you don't need extension anymore : its added features are list filters, embedded lists and role-based permissions.

EasyAdmin now has list filters and role permissions (maybe not the same coverage as the extension). So there is nearly only embedded lists that value the use of the extension. This is an announced future feature for EasyAdmin, so the extension will soon be useless (and that's a good point). I have no time to work on compatibility, I would rather wait for embedded lists in EasyAdmin and migrate...

So if you want to remove the top-limit, you can fork and work on it !

**Comment by rodrigobb** [comment]
Everything is clear now. I'm actuallly using embedded lists so I'm afraid I won't be able to update.

Thanks for all the work anyway

**Comment by alterphp** [comment]
Hi !

I finally managed to make extension compatible with installation of EasyAdmin 2.2.0 and upper, and Symfony 5 as well. But ... there is still no compatibility between the implementations of list filters : use only extension implementation in order to use embedded lists (this is a "per entity" constraint). EasyAdmin dynamic list filters are not compatible with query filters (nor embedded lists).

I've added some advertisements on the README, please let me know what you think about it (how to make it more clear, more visible ?).

Can you test this PR and tell me ? #166 @rodrigobb @TomasVotruba @qcho

**Comment by alterphp** [comment]
Has anybody tested this branch ?

**Comment by TomasVotruba** [comment]
I'm waiting for https://github.com/stof/StofDoctrineExtensionsBundle/pull/403 to be out :/

**Comment by alterphp** [comment]
Ok, major version upgrade with many bundles is always a long way to go !

**Comment by TomasVotruba** [comment]
It's painful, but well said :smile:

**Comment by qcho** [comment]
Couldn' test it because of the slash:
```
 composer update
                                                                                                          
  [UnexpectedValueException]                                                                              
  Could not parse version constraint dev/ea-2.2-and-upper: Invalid version string "dev/ea-2.2-and-upper"  
                
```
I changed composer from my branch:
```diff
diff --git a/app/composer.json b/app/composer.json
index 45905d9..8c7992f 100644
--- a/app/composer.json
+++ b/app/composer.json
@@ -3,17 +3,11 @@
     "license": "proprietary",
     "name": "qbic/bonanza",
     "description": "Sociedad de Capitalización y Renta",
-    "repositories": [
-        {
-            "type": "vcs",
-            "url": "https://github.com/qcho/EasyAdminExtensionBundle/"
-        }
-    ],
     "require": {
         "php": "^7.1.3",
         "ext-ctype": "*",
         "ext-iconv": "*",
-        "alterphp/easyadmin-extension-bundle": "dev-filters-compat",
+        "alterphp/easyadmin-extension-bundle": "dev/ea-2.2-and-upper",
         "doctrine/annotations": "^1.6",
         "easycorp/easyadmin-bundle": "^2.3",
         "guzzlehttp/guzzle": "~6.0",
```

**Comment by alterphp** [comment]
Hi @qcho !

you have to prefix the branch name by `dev-` like following :

```
{
    "require": {
        "alterphp/easyadmin-extension-bundle": "dev-dev/ea-2.2-and-upper as 2.2.0"
    }
}
```

**Comment by qcho** [comment]
It seems to work fine for me. Just needed to change the `filters` usages for `ext_filters` when including embedded lists manually in twig

**Comment by alterphp** [comment]
Fixed by #166 => tagged as `v3.0.0`
